### ISSUE TITLE
Regression: wildcard in type of generalizable variable becomes irrelevant

### ISSUE THREAD

**Issue by jespercockx** [agda/agda#5363]
Here's a minimal example:
```agda
open import Agda.Primitive

variable
  A : Set _

levelOf : A → Level
levelOf {a} _ = a
```
On current master, this raises an error:
```
Variable a is declared irrelevant, so it cannot be used here
when checking that the expression a has type Level
```
This used to work in 2.6.1, and I believe up to very recently. It is causing the build of my PR https://github.com/agda/agda/pull/5354 to fail.

It seems to be a problem specific to universe levels, as the following variant is still accepted:
```agda
postulate
  A : Set
  B : A → Set
  T : ∀ {x} → B x → Set

variable
  b : B _

aOf : T b → A
aOf {a} _ = a
```

**Comment by jespercockx** [comment]
Looking at the debug output, the inferred type of `levelOf` is actually `..{A.1 : Level} {A : Set A.1} → A → Level`, so somehow shape irrelevance managed to sneak its way into the type somehow.

**Comment by jespercockx** [comment]
This was introduced by the fix for #5058 in https://github.com/agda/agda/commit/2f896975db85287b13db5461428c8abcbfae1513. That patch is doing the right thing for metas that were created from explicitly mentioned generalizable variables, but not for metas created from the type of other generalizable metas: these should be generalized using the default modality.
